bonfire/Modules/Tools/Controllers/LogsController.php

```diff
@@ -0,0 +1,259 @@
+        'CRITICAL' => 'fas fa-exclamation-triangle',
+        'DEBUG' => 'fas fa-bug',
+        'ALL'   => 'fas fa-minus',
+    ];
+
+    private static $levelClasses = [
+        'INFO'     => 'info',
+        'ERROR'    => 'danger',
+        'CRITICAL' => 'danger',
+        'DEBUG'    => 'warning',
+        'ALL'      => 'muted',
+    ];
+
+    const MAX_LOG_SIZE = 52428800; //50MB
+    const MAX_STRING_LENGTH = 350; //300 chars
+
+    const LOG_LINE_START_PATTERN = "/((INFO)|(ERROR)|(CRITICAL)|(DEBUG)|(ALL))[\s\-\d:\.\/]+(-->)/";
+    const LOG_DATE_PATTERN = ["/^((ERROR)|(CRITICAL)|(INFO)|(DEBUG)|(ALL))\s\-\s/", "/\s(-->)/"];
+    const LOG_LEVEL_PATTERN = "/^((ERROR)|(CRITICAL)|(INFO)|(DEBUG)|(ALL))/";
+
+    /**
+     * Displays all logs.
+     *
+     * @return string
+     */
+    public function index()
+    {
+        
+      if (isset($_POST['delete'])) {
+
+          $checked = $_POST['checked'];
+          $numChecked = count($checked);
+
+          if (is_array($checked) && $numChecked) {
+
+              foreach ($checked as $file) {
+                  @unlink($this->logs_path . $file);
+              }
+
+            return redirect()->to(ADMIN_AREA.'tools/logs')->with('message', lang('Logs.delete_success'));
+
+          }
+
+        } elseif (isset($_POST['delete_all'])) {
+
+          helper('filesystem');
+
+          if (delete_files($this->logs_path)) {
+
+              // Restore the index.html file.
+              @copy(APPPATH . '/index.html', "{$this->logs_path}index.html");
+              return redirect()->to(ADMIN_AREA.'tools/logs')->with('message', lang('Logs.delete_all_success'));
+
+
+          } else {
+              return redirect()->to(ADMIN_AREA.'tools/logs')->with('error', lang('Logs.delete_error'));
+          }
+      }
+
+        global $app;
+        $db = db_connect();
+
+
+        helper('form');
+
+        $uri = new \CodeIgniter\HTTP\URI();
+
+        $pager = \Config\Services::pager();
+
+        $page = isset($_GET['page']) ? $_GET['page'] : 1;
+
+        // Load the Log Files.
+        $logs = get_filenames($this->logs_path);
+
+        unset($logs[0]);
+
+        $total = count($logs);
+
+        $limit = 15;
+
+        $offset = ($page > 1)? ($page-1)*$limit: 0;
+
+        $pager->makeLinks($page,$limit,$total);
+
+        $logs = array_slice($logs, $offset, $limit);
+
+        return $this->render($this->viewPrefix .'logs', [
+            'logs' => $logs,
+            'pager' => $pager,
+        ]);
+    }
+
+    /**
+     * Show the contents of a single log file.
+     *
+     * @param string $file The full name of the file to view (including extension).
+     *
+     * @return void
+     */
+    public function view(string $file = '')
+    {
+        if (empty($file)) {
+            Template::set_message(lang('logs_view_empty'), 'danger');
+            redirect(SITE_AREA . '/developer/logs');
+        }
+
+        helper('form');
+
+        $pager = \Config\Services::pager();
```

Comment:
I prefer using the helper method here. It's possible it will gain additional powers in the future. 

```suggestion
        $pager = service('pager');
```